rotkehlchen/chain/ethereum/modules/balancer/decoder.py

```diff
@@ -0,0 +1,296 @@
+            to_counterparty=CPT_BALANCER_V2,
+            to_notes=None,
+            extra_data={
+                'from_token': from_token_address,
+                'amount_in': amount_in,
+            },
+        )
+
+        # When ETH is swapped it is wrapped to WETH and the ETH transfer happens before the SWAP
+        # event, we need to detect it we haven't done it yet.
+        if len(action_items) == 0 and from_token_address == self.weth.evm_address:
+            # when swapping eth the transfer event appears before the V2_SWAP event so we need
+            # to check if the asset swapped was ETH or not.
+            amount_of_eth = asset_normalized_value(
+                amount=amount_in,
+                asset=self.eth,
+            )
+            for event in decoded_events:
+                if (
+                    event.asset == A_ETH and event.balance.amount == amount_of_eth and
+                    event.event_type == HistoryEventType.SPEND and
+                    event.event_subtype == HistoryEventSubType.NONE
+                ):
+                    event.event_type = HistoryEventType.TRADE
+                    event.event_subtype = HistoryEventSubType.SPEND
+                    event.notes = f'Swap {event.balance.amount} {self.eth.symbol} in Balancer v2 from {event.location_label}'  # noqa: E501
+                    event.counterparty = CPT_BALANCER_V2
+
+        return None, [action_item]
+
+    def _maybe_enrich_balancer_v2_transfers(  # pylint: disable=no-self-use
+            self,
+            token: 'EvmToken',  # pylint: disable=unused-argument
+            tx_log: EvmTxReceiptLog,
+            transaction: EvmTransaction,
+            event: HistoryBaseEntry,
+            action_items: list[ActionItem],
+            all_logs: list[EvmTxReceiptLog],  # pylint: disable=unused-argument
+    ) -> bool:
+        """
+        Enrich tranfer transactions to account for swaps in balancer v2 protocol.
+        May raise:
+        - UnknownAsset
+        - WrongAssetType
+        """
+        if action_items is None or len(action_items) == 0 or transaction.to_address != VAULT_ADDRESS:  # noqa: E501
+            return False
+
+        if action_items[-1].extra_data is None:
+            return False
+
+        asset = event.asset.resolve_to_evm_token()
+
+        if (
+            isinstance(action_items[-1].asset, EvmToken) is False or
+            action_items[-1].asset.evm_address != tx_log.address or  # type: ignore[attr-defined]  # noqa: E501 It fails to detect the previous comprobation
+            action_items[-1].amount != event.balance.amount
+        ):
+            return False
+
+        event.counterparty = CPT_BALANCER_V2
+        event.event_type = HistoryEventType.TRADE
+        if asset == event.asset:
+            event.event_subtype = HistoryEventSubType.RECEIVE
+            event.notes = f'Receive {event.balance.amount} {asset.symbol} in Balancer v2 from {event.location_label}'  # noqa: E501
```

Comment:
This has remained unchanged and my question has not been answered. Please try to appreciate the time I am putting in the reviews **and read them**

https://github.com/rotki/rotki/pull/5197#discussion_r1051653578